imputation/src/mean/mean_test.go

```diff
@@ -0,0 +1,163 @@
+package main
+
+import (
+    "testing"
+)
+
+func TestMean(t *testing.T) {
+    // test 1 empty series
+    series1 := make(map [string]float64)
+    want := series1
+    got := fillMean(series1)
+    if  len(got) != len(want) {
+        t.Errorf("Size mismatch: fillMean() = %v, want %v", got, want)
+    }
+    
+    for k,v := range got{
+        if vw, ok := want[k]; !ok || vw != v {
+            t.Errorf("Value mismatch: fillMean() = %v, want %v", got, want)
+        }
+    } 
+    
+    //test 2 small series
+    series2 := make(map [string]float64)
+    series2["2003"] = 42.5
+    want = series2
+    got = fillMean(series2)
+    if  len(got) != len(want) {
+        t.Errorf("Size mismatch: fillMean() = %v, want %v", got, want)
+    }
+    
+    for k,v := range got{
+        if vw, ok := want[k]; !ok || vw != v {
+            t.Errorf("Value mismatch: fillMean() = %v, want %v", got, want)
+        }
+    } 
+    
+    
+    //test 3 year gap
+    series3 := make(map [string]float64)
+    series3["2003"] = 42.5
+    series3["2002"] = 18.9
+    series3["1998"] = 28
+    
+    want = make(map [string]float64)
+    want["2003"] = 42.5
+    want["2002"] = 18.9
+    want["1998"] = 28
+   
+    
+    var mean float64 = 0
+    for _,v := range series3{
+        mean += v
+    }
+    mean = mean/float64(len(series3))
+    
+    want["2000"] = mean
+    want["2001"] = mean
+    want["1999"] = mean
+    
+    got = fillMean(series3)
+    if  len(got) != len(want) {
+        t.Errorf("Size mismatch: fillMean() = %v, want %v", got, want)
+    }
+    
+    for k,v := range got{
+        if vw, ok := want[k]; !ok || vw != v {
+            t.Errorf("Value mismatch: fillMean() = %v, want %v", got, want)
+        }
+    } 
+    
+    
+    //test 4 month gap
+    series4 := make(map [string]float64)
+    series4["2003-02"] = 42.5
+    series4["2002-02"] = 18.9 //edge case : 2002-01 fails currently
+    series4["2003-04"] = 28
+    want = make(map [string]float64)
+    want["2003-02"] = 42.5
+    want["2002-02"] = 18.9
+    want["2003-04"] = 28
+   
+    
+    mean = 0
+    for _,v := range series4{
+        mean += v
+    }
+    mean = mean/float64(len(series4))
```

Comment:
Let's define a function to compute the mean for use within this file.  This follows the heuristic of writign a function if code is need twice.